DUnitX.Assert.pas

```diff
@@ -359,6 +359,23 @@ class procedure Assert.AreEqual(const expected, actual: boolean; const message:
     FailFmt(SUnexpectedErrorStr ,[BoolToStr(expected, true), BoolToStr(actual, true), message], ReturnAddress);
 end;
 
+class procedure Assert.AreEqual(const expected, actual: string; const compformat: TDUnitXComparableFormatClass; const ignoreCase: boolean; const message: string);
+begin
+  DoAssert;
+  if ignoreCase then
+  begin
+    if not SameText(expected,actual) then
+      FailStrCompare(expected, actual, compformat, message, ReturnAddress);
```

Comment:
Yes, that is correct.  The intention was NOT to add more granular capabilities to the Assert, but to be able to control the diff on the GUI side.  In other words, the format classes (as of now, and as I originally have intended) were to ONLY be used for helping you diff the failure - not change the comparison.  In my mind, your test should return the EXACT result you are expecting when it comes to strings (so if expecting CRLF, it should have CRLF), but when it comes to diffing said result, you may want the niceness of the compare tool's formatting capability.  We have a lot of XML that gets generated that can't have CRLF on it.  Diffing that without BeyondCompare formatting it is a nightmare (one long line), thus why I even allow specifying a format in the first place.  Yes, you can manually select XML in BC, but when you are working bugs quickly, the number of clicks really starts to be prohibitive.  Perhaps, down the road, the Assert could take some rules into account, but that wasn't my intention at this stage.  It is just a way for the GUI to know what the format is SUPPOSED to be and how best to diff it.